controller/Review.php

```diff
@@ -100,68 +104,88 @@ public function init(): void
 
         $params = $this->getPsrRequest()->getQueryParams();
 
-        if (isset($params['serviceCallId'])) {
-
-            /** @var DeliveryExecutionFinderService $finder */
-            $finder = $this->getServiceLocator()->get(DeliveryExecutionFinderService::SERVICE_ID);
-
-            $data = $dataBuilder->create()->build($params['serviceCallId'], $finder->getShowScoreOption($this->ltiSession->getLaunchData()));
+        try {
+            if (isset($params['serviceCallId'])) {
```

Comment:
I highly recommend to add this new code / validations inside the `DeliveryExecutionFinderService:: findDeliveryExecution` so we can test it and also we can avoid same issue in different places. I do not think the controller should do this.

There is also tests missing.

Reply:
So, its not the new code. I just added try/catch and one method. Because for now, we just need to fix that security issue.

Reply:
Not sure that we need to refactor that extension during this task

Reply:
Yes, I understand. I am referring only about the new code that throws the exception and validate if user is the owner. I understand the previous code sometimes is not possible to be extracted :)